felix/bpf-gpl/tc.c

```diff
@@ -1327,6 +1327,9 @@ int calico_tc_skb_accepted_entrypoint(struct __sk_buff *skb)
 		deny_reason(ctx, CALI_REASON_DROPPED_BY_QOS);
 		goto deny;
 	}
+	if (set_dscp(ctx) == TC_ACT_SHOT) {
```

Comment:
```suggestion
	if ((CALI_F_FROM_WEP || CALI_F_TO_HEP) && EGRESS_DSCP >= 0 && !set_dscp(ctx)) {
```

you could make the set_dscp return bool. I'd suggest name it `qos_set_dscp` will suggest to rename the rate functions to qos_* too